ts/tat-trend.ts

```diff
@@ -0,0 +1,437 @@
+    if (!assayColors[assay]) {
+      assayColors[assay] = generateColor(colorIndex++);
+    }
+    selectedGates.forEach((gate) => {
+      if (assayGroups[assay][gate]) {
+        newPlot.push({
+          x: assayGroups[assay][gate].x,
+          y: assayGroups[assay][gate].y,
+          type: "box",
+          name: `${assay}`,
+          text: assayGroups[assay][gate].text,
+          hoverinfo: "text+y",
+          hovertemplate: `
+            %{text}<br>
+            Days: %{y}<br>
+            N: %{customdata}
+          `,
+          customdata: Array(assayGroups[assay][gate].x.length).fill(
+            assayGroups[assay][gate].n
+          ),
+          boxpoints: "all",
+          jitter: 0.3,
+          pointpos: 0,
+          marker: {
+            size: 6,
+            color: colorByGate ? gateColors[gate] : assayColors[assay],
+          },
+          boxmean: true,
+          legendgroup: assay,
+          showlegend: !newPlot.some((d) => d.legendgroup === assay),
+        } as unknown as Partial<Plotly.PlotData>);
+      }
+    });
+  });
+
+  const layout: Partial<Plotly.Layout> = {
+    xaxis: {
+      title:
+        selectedGrouping === "week"
+          ? "Week"
+          : selectedGrouping === "month"
+          ? "Month"
+          : selectedGrouping === "fiscalQuarter"
+          ? "Fiscal Quarter"
+          : "Fiscal Year",
+      tickformat:
+        selectedGrouping === "week"
+          ? "%Y-W%U"
+          : selectedGrouping === "month"
+          ? "%Y-%m"
+          : selectedGrouping === "fiscalQuarter"
+          ? "%Y Q%q"
+          : "%Y",
+      categoryorder: "category ascending",
+    },
+    yaxis: {
+      title: "Days",
+      zeroline: false,
+      range: [0, undefined],
+    },
+    boxmode: "group",
+    autosize: true,
+    uirevision,
+    width: window.innerWidth - 50,
+    height: window.innerHeight - 250,
+  };
+
+  return { newPlot, layout };
+}
+
+function newPlot(
+  selectedGrouping: string,
+  jsonData: any[],
+  selectedGates: string[],
+  selectedDataType: string
+) {
+  const trendReportContainer = document.getElementById("trendReportContainer");
+
+  if (trendReportContainer) {
+    const plotContainer = document.getElementById("plotContainer");
+    if (plotContainer) {
+      plotContainer.textContent = "";
+    }
+
+    const { newPlot, layout } = plotData(
+      jsonData,
+      selectedGrouping,
+      selectedGates,
+      selectedDataType
+    );
+    Plotly.newPlot("plotContainer", newPlot, layout);
+
+    window.addEventListener("resize", () => {
+      Plotly.relayout("plotContainer", {
+        width: window.innerWidth,
+        height: window.innerHeight,
+      });
+    });
+  }
+}
+
+function updatePlot(
+  selectedGrouping: string,
+  jsonData: any[],
+  selectedGates: string[],
+  selectedDataType: string
+) {
+  const trendReportContainer = document.getElementById("trendReportContainer");
+
+  if (trendReportContainer) {
```

Comment:
This being undefined should be an error, right? `html-utils` has a `getRequiredElementById` function for this